hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenClaimAirdropHandler.java

```diff
@@ -42,10 +53,46 @@ public TokenClaimAirdropHandler() {
     }
 
     @Override
-    public void preHandle(@NonNull PreHandleContext context) throws PreCheckException {}
+    public void preHandle(@NonNull PreHandleContext context) throws PreCheckException {
+        requireNonNull(context);
+        final var op = context.body().tokenClaimAirdrop();
+        final var pendingAirdrops = op.pendingAirdrops();
+
+        if (pendingAirdrops.isEmpty()) {
+            throw new PreCheckException(EMPTY_PENDING_AIRDROP_ID_LIST);
+        }
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+
+        for (final var pendingAirdrop : pendingAirdrops) {
+            AccountID receiverId = pendingAirdrop.receiverIdOrThrow();
+            Account account = accountStore.getAccountById(receiverId);
+            requireNonNull(account);
+            context.requireKey(account.key());
+        }
+    }
 
     @Override
-    public void pureChecks(@NonNull TransactionBody txn) throws PreCheckException {}
+    public void pureChecks(@NonNull TransactionBody txn) throws PreCheckException {
+        requireNonNull(txn);
+
+        final var op = txn.tokenClaimAirdrop();
+        requireNonNull(op);
+
+        final List<PendingAirdropId> pendingAirdrops = op.pendingAirdrops();
+        if (pendingAirdrops.isEmpty()) {
+            throw new PreCheckException(EMPTY_PENDING_AIRDROP_ID_LIST);
+        }
+
+        if (pendingAirdrops.size() > 10) {
+            throw new PreCheckException(MAX_PENDING_AIRDROP_ID_EXCEEDED);
+        }
+
+        final int numAirdrops = pendingAirdrops.size();
+        final Set<PendingAirdropId> uniqueAirdrops = Set.copyOf(pendingAirdrops);
+        if (numAirdrops != uniqueAirdrops.size()) {
+            throw new PreCheckException(PENDING_NFT_AIRDROP_ALREADY_EXISTS);
```

Comment:
The protobuf decl of this response code doesn't have a Javadoc, unfortunately.  If it did it would be nice if it said that this response code can also mean "duplicate nft airdrops specified in same tx".

Reply:
@povolev15 I wonder if this check is needed at all. On the `TokenAirdropTransaction` we are throwing `PENDING_NFT_AIRDROP_ALREADY_EXISTS` when we try to create a new airdrop but it already exists in the store/state not if we have duplicated airdrops in a single transaction.

In the case of a claim, if we try to **claim** an airdrop twice in a single transaction I would think that the first **claim** will be successful and the second one will throw that the airdrop doesn't exist - which sounds logical to me. What do you think?

Reply:
For the `AirdropTransaction` we have the same check(https://github.com/hashgraph/hedera-services/pull/14279). You can see it in `CryptoTransferValidator.validateNftTransfers`:
```
validateFalsePreCheck(
        !nftIds.isEmpty() && nftIds.contains(nftTransfer.serialNumber()), INVALID_ACCOUNT_AMOUNTS);
```
And we are throwing `INVALID_ACCOUNT_AMOUNTS`. In my opinion, this is not the most precise error code but we are reusing the validations from CryptoTransfer. So I think we should be consistent with that and update the response code here from `PENDING_NFT_AIRDROP_ALREADY_EXISTS` to `INVALID_ACCOUNT_AMOUNTS`. What do you think?